tb-dev-challenge/tb-dev-challenge/ItemAddEditViewController.swift

```diff
@@ -0,0 +1,182 @@
+            deleteItemButton.isHidden = false
+            deleteItemButton.addTarget(self, action: #selector(deleteItem), for: .touchUpInside)
+            menuItemNameTextField.text = itemSelected?.itemName
+            menuItemPriceTextField.text = "\(itemSelected?.itemPrice ?? 0.00)"
+            itemImagePreview.image = itemSelected?.itemImage?.toImage()
+        }
+
+        self.navigationItem.rightBarButtonItem = UIBarButtonItem(barButtonSystemItem: .save, target: self, action: #selector(createNewItemForGroup))
+        // MARK: Upload Button
+        let imageItemUploadButton = UIButton.init(type: .system)
+
+        imageItemUploadButton.frame = CGRect(x: 200,
+                                             y: 400,
+                                             width: 200,
+                                             height: 60)
+        imageItemUploadButton.setTitle("Upload Item Image",
+                                       for: .normal)
+
+        imageItemUploadButton.setTitleColor(.systemBlue,
+                                            for: .normal)
+        imageItemUploadButton.addTarget(self,
+                                        action: #selector(imageItemUploadAction),
+                                        for: .touchUpInside)
+        self.view.addSubview(imageItemUploadButton)
+
+        menuItemPriceTextField.delegate = self
+        setupView()
+
+    }
+
+    @objc func deleteItem() {
+        let menuItemToDelete = self.itemSelected
+        self.context.delete(menuItemToDelete!)
+
+        do {
+            try self.context.save()
+        } catch {
+            print("Failed to save data")
+        }
+        _ = navigationController?.popViewController(animated: true)
+    }
+
+    @objc func createNewItemForGroup() {
+
+        if itemSelected == nil {
+            let itemName = menuItemNameTextField.text
+            let itemPriceString = menuItemPriceTextField.text ?? ""
+            let itemPrice = Double(itemPriceString)
+            let itemImage = imageSelected
+
+            if let addNewItem = MenuItems(itemImage: itemImage, itemName: itemName, itemPrice: itemPrice, itemDateCreated: Date()) {
+                menuGroup?.addToOfMenuItem(addNewItem)
+
+                do {
+                    try addNewItem.managedObjectContext?.save()
+                    _ = navigationController?.popViewController(animated: true)
+                } catch {
+                    print("Menu item failed to save!")
+                }
+            }
+        } else {
+
+            let itemName = menuItemNameTextField.text
+            let itemPriceString = menuItemPriceTextField.text ?? ""
+            let itemPrice = Double(itemPriceString)
+            let itemImage = imageSelected!
+            do {
+                itemSelected?.setValue(itemName, forKey: "itemName")
+                itemSelected?.setValue(itemImage, forKey: "itemImage")
+                itemSelected?.setValue(itemPrice, forKey: "itemPrice")
```

Comment:
MenuItem could have its own CoreData entity and NSManagedObject subtype

Reply:
just saw the subtype below, if there's a MenuItems type we can remove this key-value style

Reply:
might need to pair with you to understand this.